src/python/pants/backend/javascript/dependency_inference/rules.py

```diff
@@ -101,21 +121,107 @@ async def map_candidate_node_packages(
     )
 
 
+@dataclass(frozen=True)
+class InferenceMetadataRequest:
+    imports: PackageJsonImports
+    config: TSConfig | None
+
+
 @rule
-async def prepare_inference_metadata(imports: PackageJsonImports) -> InferenceMetadata:
+async def prepare_inference_metadata(req: InferenceMetadataRequest) -> InferenceMetadata:
```

Comment:
(Totally minor observation about existing code: this doesn't do anything `async`, so could be a non-`@rule` non-`async` normal function, potentially, and thus avoid needing the new `InferenceMetadataRequest` type etc. since it can just take multiple args)